### PR TITLE
Fix TID validation rejecting valid TIDs starting with a-j

### PR DESCRIPTION
**Pull request by jacoblapenna**
Hi! Great work in this SDK!

While writing spec-conformance tests for an AT Protocol project I'm working on, I used this library's validators as a cross-check and hit a case where we disagreed on a valid TID. I believe the first-character check in validate_tid has a subtle bug:

`if not TID_RE.match(v) or (ord(v[0]) & 0x40):`

The intent (per the [TID spec](https://atproto.com/specs/tid)) is that the high bit of the decoded 64-bit value must be 0, which restricts the first character to 234567abcdefghij. But ord(v[0]) & 0x40 tests bit 6 of the character's ASCII encoding, and every lowercase letter has that bit set — so all spec-valid TIDs starting with a–j are rejected:

```
TypeAdapter(Tid).validate_python('a222222222222', context={'strict_string_format': True})
# ValidationError, but this is a valid TID
```

This slipped past CI because the interop test files don't include a letter-leading valid TID. This is true of the upstream bluesky-social/atproto vectors too (I checked; the vendored copies here are identical to upstream), so the suite was rightfully green. I'm thinking of proposing the missing vector upstream as well.

The fix encodes the constraint in the regex itself, matching the reference implementation in [@atproto/syntax tid.ts](https://github.com/bluesky-social/atproto/blob/main/packages/syntax/src/tid.ts) character-for-character (`/^[234567abcdefghij][234567abcdefghijklmnopqrstuvwxyz]{12}$/`), and removes the ASCII-bit check. Added parametrized tests for valid first characters (a, j) and invalid ones (k, z, 1, 8). The valid-case tests fail on main and pass with this change. Full suite: 445 passed.

Happy to adjust anything for naming, test placement, whatever fits your conventions best. Thanks!

### CODE REVIEW

**Comment by jacoblapenna** [comment]
Follow-up on the interop gap mentioned above: I've proposed the missing vectors upstream in https://github.com/bluesky-social/atproto/pull/5243. It adds three letter-first valid TIDs (`a222222222222`, `j222222222222`, `azzzzzzzzzzzz`) to `tid_syntax_valid.txt` — the accepted side of the boundary that let this bug pass the interop suite. If it lands, this SDK's interop-driven tests will cover the a–j first-character range automatically, and the fix in this PR is exactly what makes them pass.

File: packages/atproto_client/models/string_formats.py [hunk 1]
```diff
@@ -424,9 +424,10 @@ def validate_tid(v: str, _: ValidationInfo) -> str:
 
     - Exactly 13 characters
 
-    - Only lowercase letters and numbers 2-7
+    - Only lowercase letters and numbers 2-7 (base32-sortable alphabet)
 
-    - First byte's high bit (0x40) must be 0
+    - First character must be one of 234567abcdefghij so that the high bit
```
Comment (MarshalX) [inline packages/atproto_client/models/string_formats.py]:
```suggestion
    - First character must be one of 234567abcdefghij so that the 65-bit base32 encoding fits in a 64-bit integer
```

File: tests/test_atproto_client/models/tests/test_string_formats.py [hunk 2]
```diff
@@ -124,6 +124,24 @@ def test_string_format_validation_with_valid(field_name: str, valid_value: str)
     assert validated == valid_value
 
 
+@pytest.mark.parametrize('valid_tid', ['a222222222222', 'j222222222222', 'aaaaaaaaaaaaa', '3jzfcijpj2z2a'])
```
Comment (MarshalX) [inline tests/test_atproto_client/models/tests/test_string_formats.py]: `3jzfcijpj2z2a` duplicate existing interop-file case

File: tests/test_atproto_client/models/tests/test_string_formats.py [hunk 3]
```diff
@@ -124,6 +124,24 @@ def test_string_format_validation_with_valid(field_name: str, valid_value: str)
...
+def test_tid_valid_first_letters(valid_tid: str) -> None:
+    """Test that TIDs starting with a-j (high bit of the 64-bit value is 0) are accepted.
+
+    The interop test files do not cover valid TIDs starting with a letter.
+    """
+    TidTypeAdapter = TypeAdapter(string_formats.Tid)
+    assert TidTypeAdapter.validate_python(valid_tid, context={_OPT_IN_KEY: True}) == valid_tid
+
+
+@pytest.mark.parametrize('invalid_tid', ['k222222222222', 'zzzzzzzzzzzzz', '1222222222222', '8222222222222'])
```
Comment (MarshalX) [inline tests/test_atproto_client/models/tests/test_string_formats.py]: `zzzzzzzzzzzzz` duplicate existing interop-file case

File: tests/test_atproto_client/models/tests/test_string_formats.py (hunk 2 above)
Comment (MarshalX) [inline tests/test_atproto_client/models/tests/test_string_formats.py]: Let's add `2222222222222` test case here (lowest-boundary first char)

**Review by MarshalX — COMMENTED** [review]
Thanks for catching this and for the fix! I left a few inline comments, all minor

Happy to merge once the docstring wording is adjusted

**Comment by jacoblapenna** [comment]
All updated per comments.
